lib/order.rb

```diff
@@ -8,11 +14,46 @@ def initialize(id, products)
     end
 
     def total
-      # TODO: implement total
+      ((@products.values.sum)*1.075).round(2)
     end
 
     def add_product(product_name, product_price)
-      # TODO: implement add_product
+      if @products.keys.include?(product_name)
+        return false
+      else
+        @products[product_name] = product_price
+        return true
+      end
+    end
+
+    def self.all
+      all_orders = []
+      CSV.open(FILE_NAME, 'r').each do |num|
+        new_item_hash = {}
+        id = num[0].to_i
+        num[1].split(";").each do |item|
+          new_item = item.split(":")
+          key = new_item[0]
+          value = new_item[1].to_f
+          new_item_hash[key] = value
+        end
+        new_order = Order.new(id, new_item_hash)
+        all_orders << new_order
+      end
+      return all_orders
     end
+
+    def self.find(good_id)
+      return_value = "Sorry, that order is not in our records"
```

Comment:
A better return value would be `nil` for not finding it, as `nil` means nothing.